pdns/lua-record.cc

```diff
@@ -729,6 +727,24 @@ static void setupLuaRecords()
       return std::string("unknown");
     });
 
+  lua.writeFunction("filterForward", [](string address, std::vector<std::pair<int, string>> masks) {
+      NetmaskGroup nmg;
+      ComboAddress ca(address);
+
+      for (const auto& mask : masks) {
```

Comment:
it could make sense to store an NMG per LUA record so there's no need to recreate is every lookup.

Reply:
`filterForward` now takes an NMG instead of a table, which makes caching possible (with some extra typing on the user's end)